File: trunk/managed/Warp3D/Warp3DVB/warp_Light.vb
' Created by: X
' http://www.createdbyx.com/
' ----------------------
' C# to VB.NET conversion of this code was done by hand and with the assistance of the 
' Convert C# to VB .NET application availible at http://www.kamalpatel.net/ConvertCSharp2VB.aspx
' ConvertCSharp2VB utility was developed by: Kamal Patel (http://www.KamalPatel.net) 
' ----------------------
' Thanks to Alan Simes ( http://alansimes.blogdns.net/ ) for creating the Warp3D library.
' http://alansimes.blogdns.net/cs/files/default.aspx
' ----------------------

Imports System

Namespace Rednettle.Warp3D
    '/ <summary>
    '/ Summary description for warp_Light.
    '/ </summary>
    Public Class warp_Light
        ' F I E L D S

        Public v As warp_Vector  'Light direction()
        Public v2 As warp_Vector  'projected Light direction 
        Public diffuse As Integer = 0
        Public specular As Integer = 0
        Public highlightSheen As Integer = 0
        Public highlightSpread As Integer = 0

        Dim matrix2 As warp_Matrix

        Private Sub New()
        End Sub

        Public Sub New(ByVal direction As warp_Vector)
            v = direction.getClone()
            v.normalize()
        End Sub

        Public Sub New(ByVal direction As warp_Vector, ByVal diffuse As Integer)
            v = direction.getClone()
            v.normalize()
            Me.diffuse = diffuse
        End Sub

        Public Sub New(ByVal direction As warp_Vector, ByVal color As Integer, ByVal highlightSheen As Integer, ByVal highlightSpread As Integer)
            v = direction.getClone()
            v.normalize()
            Me.diffuse = color
            Me.specular = color
            Me.highlightSheen = highlightSheen
            Me.highlightSpread = highlightSpread
        End Sub

        Public Sub New(ByVal direction As warp_Vector, ByVal diffuse As Integer, ByVal specular As Integer, ByVal highlightSheen As Integer, ByVal highlightSpread As Integer)
            v = direction.getClone()
            v.normalize()
            Me.diffuse = diffuse
            Me.specular = specular
            Me.highlightSheen = highlightSheen
            Me.highlightSpread = highlightSpread
        End Sub

        Public Sub project(ByVal m As warp_Matrix)
            matrix2 = m.getClone()
            matrix2.transform(m)
            v2 = v.transform(matrix2)
        End Sub
    End Class
End Namespace

File: trunk/managed/Warp3Dmod/warp_Light.cs
using System;

namespace Warp3D
{
	/// <summary>
	/// Summary description for warp_Light.
	/// </summary>
	public class warp_Light
	{
		// F I E L D S

		public warp_Vector v; //Light direction
		public warp_Vector v2; //projected Light direction
		public int diffuse = 0;
		public int specular = 0;
		public int highlightSheen = 0;
		public int highlightSpread = 0;

		warp_Matrix matrix2;

		private warp_Light()
			// Default constructor not accessible
		{
		}

		public warp_Light(warp_Vector direction)
		{
			v = direction.getClone();
			v.normalize();
		}

		public warp_Light(warp_Vector direction, int diffuse)
		{
			v = direction.getClone();
			v.normalize();
			this.diffuse = diffuse;
		}

		public warp_Light(warp_Vector direction, int color, int highlightSheen, int highlightSpread)
		{
			v = direction.getClone();
			v.normalize();
			this.diffuse = color;
			this.specular = color;
			this.highlightSheen = highlightSheen;
			this.highlightSpread = highlightSpread;
		}

		public warp_Light(warp_Vector direction, int diffuse, int specular,int highlightSheen, int highlightSpread)
		{
			v = direction.getClone();
			v.normalize();
			this.diffuse = diffuse;
			this.specular = specular;
			this.highlightSheen = highlightSheen;
			this.highlightSpread = highlightSpread;
		}

		public void project(warp_Matrix m)
		{
			matrix2 = m.getClone();
			matrix2.transform(m);
			v2 = v.transform(matrix2);
		}
	}
}
